models/UserModel.ts

```diff
@@ -121,6 +122,7 @@ export class UserModel extends BaseEntity {
   }
 
   public getFullUserProfile(): PrivateProfile {
+    const resumes = await ResumeRepository.findAllByUser(this); // TODO: somehow get to userresume
     return {
```

Comment:
Here's my idea: let's not perform the resume lookup in `getFullUserProfile()`, and instead let's create a new function `async getResumes()` that does the lookup and populates the `user.resumes` field. Then, in the getCurrentUser() route, we can call this function and then return `user.getFullUserProfile()` in the API response.

The logic behind this is that there's only one route that would ever *need* to return resumes, which is the getCurrentUser() route. All the other calls to getFullUserProfile() don't actually need the resumes to be populated, which is why I'm hesitant to add an extra query to that function.

Let's still return `resumes: user.resumes` in the getFullUserProfile() return object so that the field is not undefined, but I don't necessarily think it needs to be populated, and if the caller needs to populate it, they can call `getResumes()` before it. Let's also leave a comment on the `getFullUserProfile()` function mentioning this.

Feel free to follow up if there's any questions with the approach

Reply:
Is there a case where calling `getFullUserProfile` actually does not want to have the resume field populated even though it was populated with the new `getResume` function?